### PR TITLE
feat: Save and render content for lexical editor - page builder

### PR DESCRIPTION
**Pull request by mihajlovco**
With this PR we want to allow content written with Lexical editor to be saved in DB. _Heading_ and _Paragraph_ components will use Lexical editor, other components like _List_ and _Quote_ are legacy components and will be removed from the sidebar.

`Note:` With this PR we still support the content of the legacy _List_ and _Quote_ components only for editing already existing content but not for adding new content. Users in the future will use the _Paragraph_ component with the Lexical editor to add a list and quote.

## General Changes
- Improved temporary `lexical-editor-pb`  folder structure
- updated `useVariable` hook with generic type

### Rendering the legacy and lexical content 
- New component for rendering lexical content LexicalHtmlRenderer.
Support for the legacy content for Heading, Paragraph, List, and Quote components content

https://user-images.githubusercontent.com/82515066/221182025-22e3cc7d-9e18-4d5e-9d1d-14d39fc5b4e2.mov

### Select and drop components to the page editor
- Support for Lexical Heading and Paragraph components 
- Legacy List and Quote components are removed from the sidebar
- Added LexicalText component 
- Added PeTextPlugin - allows to edit and save content with a lexical editor for Headings and Paragraph components 
- HeadingPlugin - renders heading content with the lexical editor
- ParagraphPlugin - renders paragraph content with the lexical editor
- check based on data, if data is not in `lexical editor` state in the case legacy medium editor will be loaded 

https://user-images.githubusercontent.com/82515066/221181212-21161a86-7b64-488c-ba2a-1cb1ddc27464.mov

### Support for Page Builder Blocks - edit and unlink

https://user-images.githubusercontent.com/82515066/221181193-44c00812-0769-4872-b18c-ce75fe4fb9ba.mov

### Lexical editor upgrades
- Support for size styles: width and height
- Added focus prop
- Added value updates 
- Added onBlur event
-  Fixed z-index for lexical-editor toolbar 
- set to 30 to support dialog window for the block editor
- Check if the value is a valid JSON object when it passed to the Lexical editor
- Refactor rename - EditorStateJSONState to LexicalValue type
- Refactor rename - getEmpyEditorStateJSONState to generateEmptyLexicalValue

Check how focus and size work:

https://user-images.githubusercontent.com/82515066/221183495-c201251e-6003-458f-be9c-610a9ec223b0.mov

## How Has This Been Tested?
Manually.

### CODE REVIEW

File: packages/lexical-editor-pb/src/plugins/PeTextPlugin.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import get from "lodash/get";
+import { applyFallbackDisplayMode } from "@webiny/app-page-builder/editor/plugins/elementSettings/elementSettingsUtils";
+import { isParagraphTag } from "~/utils/isParagraphTag";
+import { isHeadingTag } from "~/utils/isHeadingTag";
+
+const DATA_NAMESPACE = "data.text";
+export const PeTextPlugin = createComponentPlugin(PeText, Original => {
+    return function PbTextPlugin({ elementId, tag: customTag, mediumEditorOptions }): JSX.Element {
+        const element = useRecoilValue(elementWithChildrenByIdSelector(elementId));
+        const [{ displayMode }] = useRecoilState(uiAtom);
```
Comment (Pavel910) [inline packages/lexical-editor-pb/src/plugins/PeTextPlugin.tsx]: There are hooks for this, so you don't have to work with recoil directly, check this folder and simplify your code usign these hooks: https://github.com/webiny/webiny-js/tree/dev/packages/app-page-builder/src/editor/hooks

File: packages/lexical-editor-pb/tsconfig.json [hunk 2]
```diff
@@ -18,7 +22,15 @@
...
-      "@webiny/lexical-editor": ["../lexical-editor/src"]
+      "@webiny/lexical-editor": ["../lexical-editor/src"],
+      "@webiny/ui": ["../ui/src/*"],
+      "@webiny/ui": ["../ui/src"],
+      "@webiny/feature-flags": ["../feature-flags/src/*"],
+      "@webiny/feature-flags": ["../feature-flags/src"],
+      "@webiny/app-page-builder-elements": ["../app-page-builder-elements/src/*"],
+      "@webiny/app-page-builder-elements": ["../app-page-builder-elements/src"],
+      "@webiny/react-composition": ["../react-composition/src/*"],
+      "@webiny/react-composition": ["../react-composition/src"]
```
Comment (Pavel910) [inline packages/lexical-editor-pb/tsconfig.json]: These paths are wrong. Compare with the ones aboe, which are correct. Even better, run the script to generate these for you:

```
node scripts/generateTsConfigsInPackages.js
```

**Review by Pavel910 — CHANGES_REQUESTED** [review]

File: packages/lexical-editor/src/components/Editor/RichTextEditor.tsx [hunk 3]
```diff
@@ -75,21 +88,21 @@ const BaseRichTextEditor: React.FC<RichTextEditorProps> = ({
...
                 {/* External plugins and components */}
                 {children}
                 <RichTextPlugin
                     contentEditable={
-                        <div className="editor-scroller">
-                            <div className="editor" ref={onRef}>
-                                <ContentEditable style={{ outline: 0 }} />
+                        <div className="editor-scroller" style={{ ...editorSizeStyle() }}>
+                            <div className="editor" ref={onRef} style={{ ...editorSizeStyle() }}>
+                                <ContentEditable style={{ outline: 0, ...editorSizeStyle() }} />
```
Comment (Pavel910) [inline packages/lexical-editor/src/components/Editor/RichTextEditor.tsx]: Why call `editorSizeStyle()` 4 times, when you can call it once, assign to a const, and just pass that to all 4 elements?

**Review by Pavel910 — CHANGES_REQUESTED** [review]
Last change request @mihajlovco :)

File: packages/lexical-editor/src/components/Editor/RichTextEditor.tsx [hunk 4]
```diff
@@ -49,8 +60,13 @@ const BaseRichTextEditor: React.FC<RichTextEditorProps> = ({
         }
     };
 
+    const sizeStyle = {
```
Comment (mihajlovco) [inline packages/lexical-editor/src/components/Editor/RichTextEditor.tsx:63]: @Pavel910 I directly assigned the styles to the variable without calling any method.

**Review by Pavel910 — APPROVED** [review]
Ok, good to be merged!

**Pull request merged**